File: java-backend/src/main/java/com/seekho/api/controller/EnquiryController.java
package com.seekho.api.controller;

import com.seekho.api.entity.Enquiry;
import com.seekho.api.service.EnquiryService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Optional;
@CrossOrigin(origins = "http://localhost:5173")
@RestController
@RequestMapping("/api/enquiry")
public class EnquiryController {

	@Autowired
	private EnquiryService enquiryService;

	@GetMapping("/{enquiryId}")
	public ResponseEntity<Enquiry> getEnquiryById(@PathVariable int enquiryId) {
		Optional<Enquiry> enquiry = enquiryService.getEnquiryById(enquiryId);
		return enquiry.map(value -> new ResponseEntity<>(value, HttpStatus.OK))
				.orElseGet(() -> ResponseEntity.notFound().build());
	}

	@GetMapping
	public ResponseEntity<List<Enquiry>> getAllEnquiries() {
		return new ResponseEntity<>(enquiryService.getAllEnquiries(), HttpStatus.OK);
	}

	@PostMapping
	public ResponseEntity<String> addEnquiry(@RequestBody Enquiry enquiry) {
		enquiryService.addEnquiry(enquiry);
		return ResponseEntity.status(HttpStatus.CREATED).body("Enquiry Added");
	}

	@PutMapping
	public ResponseEntity<String> updateEnquiry(@RequestBody Enquiry enquiry) {
		boolean isUpdated = enquiryService.updateEnquiry(enquiry);
		if (isUpdated) {
			return ResponseEntity.ok("Enquiry Updated");
		} else {
			return ResponseEntity.status(HttpStatus.NOT_FOUND)
					.body("There was a problem updating the Enquiry");
		}
	}

	@DeleteMapping("/{enquiryId}")
	public ResponseEntity<String> deleteEnquiry(@PathVariable int enquiryId) {
		enquiryService.deleteEnquiry(enquiryId);
		return ResponseEntity.ok("Enquiry Deleted");
	}



	@GetMapping("/getByStaffId/{staffId}")
	public ResponseEntity<List<Enquiry>> getByStaff(@PathVariable Long staffId) {
		List<Enquiry> enquiries = enquiryService.getByStaffId(staffId);
		if (enquiries.isEmpty()) {
			return ResponseEntity.notFound().build();
		} else {
			return new ResponseEntity<>(enquiries, HttpStatus.OK);
		}
	}


	@PutMapping("/deactivate/{enquiryId}")
	public ResponseEntity<String> deactivateEnquiry(@RequestBody String closureReasonDesc, @PathVariable int enquiryId) {
		enquiryService.deactivateEnquiry(closureReasonDesc, enquiryId);
		return ResponseEntity.ok("Enquiry Closed");
	}
}

File: dotnet-backend/Controllers/EnquiryController.cs
using dotnet_backend.Models;
using dotnet_backend.Services;
using Microsoft.AspNetCore.Mvc;

namespace dotnet_backend.Controllers
{
    [ApiController]
    [Route("api/[controller]")]
    public class EnquiryController : ControllerBase
    {
        private readonly IEnquiryService _enquiryService;

        public EnquiryController(IEnquiryService enquiryService)
        {
            _enquiryService = enquiryService;
        }

        // 1️⃣ Get Enquiry by ID
        [HttpGet("{enquiryId}")]
        public ActionResult<Enquiry> GetEnquiryById(int enquiryId)
        {
            var enquiry = _enquiryService.GetEnquiryById(enquiryId);
            if (enquiry == null)
                return NotFound();
            return Ok(enquiry);
        }

        // 2️⃣ Get All Enquiries
        [HttpGet]
        public ActionResult<IEnumerable<Enquiry>> GetAllEnquiries()
        {
            return Ok(_enquiryService.GetAllEnquiries());
        }

        // 3️⃣ Add Enquiry
        [HttpPost]
        public IActionResult AddEnquiry([FromBody] Enquiry enquiry)
        {
            _enquiryService.AddEnquiry(enquiry);
            return CreatedAtAction(nameof(GetEnquiryById), new { enquiryId = enquiry.EnquiryId }, enquiry);
        }

        // 4️⃣ Update Enquiry
        [HttpPut("{enquiryId}")]
        public IActionResult UpdateEnquiry(int enquiryId, [FromBody] Enquiry enquiry)
        {
            if (enquiryId != enquiry.EnquiryId)
                return BadRequest("ID mismatch");

            var updated = _enquiryService.UpdateEnquiry(enquiry);
            if (!updated)
                return NotFound();

            return Ok("Enquiry Updated");
        }

        // 5️⃣ Delete Enquiry
        [HttpDelete("{enquiryId}")]
        public IActionResult DeleteEnquiry(int enquiryId)
        {
            _enquiryService.DeleteEnquiry(enquiryId);
            return Ok("Enquiry Deleted");
        }

        // 6️⃣ Get by Staff ID
        [HttpGet("GetByStaffId/{staffId}")]
        public ActionResult<IEnumerable<Enquiry>> GetByStaffId(int staffId)
        {
            var result = _enquiryService.GetByStaffId(staffId);
            if (!result.Any())
                return NotFound();
            return Ok(result);
        }

        // 7️⃣ Deactivate Enquiry
        [HttpPut("Deactivate/{enquiryId}")]
        public IActionResult DeactivateEnquiry(int enquiryId, [FromBody] string closureReasonDesc)
        {
            _enquiryService.DeactivateEnquiry(closureReasonDesc, enquiryId);
            return Ok("Enquiry Closed");
        }
    }
}
